### ISSUE TITLE
Argus 2E?

### ISSUE THREAD

**Issue by jonahgcarpenter** [QuantumEntangledAndy/neolink#115]
**Describe the bug**
get this log error when trying to configure Argus 2E.  Yes, ive checked all my credentials and I was able to setup a Reolink Lumus but why is this one giving me trouble?

```
[2023-07-17T19:18:43Z INFO  neolink::mqtt::event_cam] Garage Camera: Connecting to camera at Address: ****, UID: ****
[2023-07-17T19:18:43Z INFO  neolink_core::bc_protocol] Garage Camera: Trying TCP discovery
[2023-07-17T19:18:43Z INFO  neolink_core::bc_protocol] Garage Camera: Trying local discovery
[2023-07-17T19:18:43Z INFO  neolink_core::bc_protocol] Garage Camera: Remote discovery success **** at ****:31457
[2023-07-17T19:18:43Z WARN  neolink::rtsp] Garage Camera: Retryable error: Garage Camera: Could not login to camera
    
    Caused by:
        0: IO Error: Custom { kind: Other, error: CameraTerminate }
        1: Camera terminated the connection
```

**To Reproduce**
```toml
bind = "0.0.0.0"

[[cameras]]
name = "Garage Camera"
username = "admin"
password = "****"
uid = "****"
address = "****"
  [cameras.mqtt]
  server = "****" # Address of the mqtt server
  port = 1883 # mqtt servers port
  credentials = ["****", "****"] # mqtt server login details
    [cameras.mqtt.discovery]
    topic = "homeassistant"
    features = ["motion", "camera", "battery", "reboot"]
```

**Versions**
NVR software: HAS
Neolink software: 0.0.6
Reolink camera model and firmware: Argus 2E v3.0.0.1813_23022300

**Comment by moskovskiy82** [comment]
Try changing your login and password to most simplest ones like 123 qwe to see if that's the problem

**Comment by jonahgcarpenter** [comment]
ive changed the pass to 123456 and still didnt work. Im assuming the username is admin but i cannot comfirm or find a place to change it?

**Comment by QuantumEntangledAndy** [comment]
CamerTerminate happens when the camera send us a disconnect command then drops the connection. There are a few reasons for this such as inactivity (to save battery), another connection requested (to save battery), or just when the signal is not strong enough. And sometimes I'm not sure why. It just seems to happen even when the connection is fine as far as I can tell.

Does this ALWAYS happen or just sometimes.

**Comment by jonahgcarpenter** [comment]
this always happens, i havent been able to get it connected

**Comment by QuantumEntangledAndy** [comment]
Interesting well I have a Argus 2E is supported for sure it is my main camera for getting UDP based cameras to work.

How are your wireshark skills, do you think you could send me a packet dump?

**Comment by QuantumEntangledAndy** [comment]
Also I notice you are using `0.0.6` which is the HAS addon. Could you tell me the actual neolink version

**Comment by jonahgcarpenter** [comment]
its neolink version 0.5.13, and ive never even heard of wireshark

**Comment by QuantumEntangledAndy** [comment]
I see, wireshark is a packet dump utilility it can be used to capture the raw packets sent over the network so I can analyse it

**Comment by QuantumEntangledAndy** [comment]
Could you update your image. I pushed a 0.5.14 update earlier today and it added a small change to the login steps. Maybe it will help

**Comment by QuantumEntangledAndy** [comment]
If I had to guess the neolink login command is not being receieved by the camera for one reason or another. Perhaps it is being lost while getting out of the docker container or by some firewall. The camera then dosen't see us sending any data and drops us. A wireshark or `nc` would tell us

**Comment by jonahgcarpenter** [comment]
id have to wait for the HAS addon update, and i can try to figure out wireshark and see

**Comment by QuantumEntangledAndy** [comment]
Can you login to the HA host? If so I can tell you the command for dumping the packets. But if your on HAOS not sure you can login to the host in the right way.

Do you know if HAOS addon made their add on pull an image OR build an image? If it was build, then reinstalling should update

**Comment by jonahgcarpenter** [comment]
Im on HAOS and i dont know the second question

**Comment by QuantumEntangledAndy** [comment]
Try to uninstall and reinstall the add on and see if that will update the neolink, if it is built locally then you will get the updated version

**Comment by jonahgcarpenter** [comment]
i still get this error on 0.5.14   

[2023-07-18T05:12:30Z INFO  neolink::mqtt::event_cam] Garage Camera: Logging in
[2023-07-18T05:12:30Z WARN  neolink::mqtt::event_cam] Camera thread error: IO Error: Custom { kind: Other, error: CameraTerminate }

**Comment by jonahgcarpenter** [comment]
i get the idea of wireshark, but i only have a wired pc to use without a wifi adapter how can i view the entirety of the networks traffic?

**Comment by QuantumEntangledAndy** [comment]
To see all network traffic you need to run something on your router (some routers have tools for this)

More easily though you should try to run wireshark or tcpdump on the ha host.

**Comment by tr1plus** [comment]
Hi - I believe I have the same issue for my Argus Eco I bought yesterday.

Below is my config(going to add mqtt ect. later, but testing with simple config):
```
[[cameras]]
name = "tuin"
username = "admin"
password = "xxx"
uid = "xxx"
#address = "192.168.50.35"
discovery = "local"
print_format = "Human"
```

same error with both address in place and commented out

```
[2023-07-20T20:36:40Z INFO  neolink_core::bc_protocol] tuin: Trying local discovery
[2023-07-20T20:36:42Z INFO  neolink_core::bc_protocol] tuin: Local discovery success xxx at 192.168.50.35:44591
[2023-07-20T20:36:42Z WARN  neolink::rtsp] tuin: Retryable error: tuin: Could not login to camera
    
    Caused by:
        0: IO Error: Custom { kind: Other, error: CameraTerminate }
        1: Camera terminated the connection
```

I have a feeling the port of the camera changes frequently.

I can provide a .cap file from pfsense, but can't upload it here. Below a screenshot.

192.168.1.30 is my docker host
192.168.50.35 is my reolink argus eco

Yes they are accross different subnets, but it's working via the reolink app(from a device in the same subnet as docker host), and it seems camera is also 'visible' to neolink, just trouble logging in. But same issue as above is happening.
Just to be sure, I'll try to put them in the same (main) subnet tomorrow in case firewall is acting up, and keep you posted.

![image](https://github.com/QuantumEntangledAndy/neolink/assets/7044850/259252f1-1def-40c8-b731-a5179acd4898)

**Comment by tr1plus** [comment]
Hi

I moved my camera to my main subnet + moved to a linux VM (instead of docker) to try to eliminate as many variables as possible.
I also tried using the original repo (thirtythreeforty).

Still getting the same error on your fork
![image](https://github.com/QuantumEntangledAndy/neolink/assets/7044850/645cb5c0-5e15-4e77-8dc1-d10f670cd775)

getting the following error on the original repo
![image](https://github.com/QuantumEntangledAndy/neolink/assets/7044850/63121b51-54e7-4905-b88b-73834f7a0c62)

Is it possible reolink preformed a firmware update which broke some functionality? I see they also released a new windows client end of June.
Since I only bought my camera mid July, it came with the latest firmware.

If you believe I should open a separate issue since I have a different camera, let me know!

I'm willing to support in packet capture (also no expert in it, but I have pfsense and can installwireshark on my linux VM).

Let me know. Appreciate the work you are doing :) Hoping to give my battery camera an RTSP stream soon ;)

**Comment by QuantumEntangledAndy** [comment]
It's possible that there is a new firmware. Some of the newer camera auto update unless you turn that off. Perhaps try sending a packet capture and I'll see what's happening. But we don't seem to be getting a error about new packet formats so would probably need a capture from the official client too

**Comment by tr1plus** [comment]
I will generate a .cap file using pfsense using neolink and a separate one using the official client. How can I share those with you since cap files are not allowed in github.

**Comment by QuantumEntangledAndy** [comment]
Just zip the capfile then it will be fine

**Comment by QuantumEntangledAndy** [comment]
While you are at it could you also trying using the official client to reduce the quality as much as possible and try again. The camera can also force a disconnect if its internal buffer is exhausted which can happen on streams that have more data to send then their is bandwidth avaliable. The official client seems to have a way to dynamically reduce the quality that I haven't been able to replicate yet.

**Comment by tr1plus** [comment]
Hi

Please see below. I hope none of the information below is too dangerous to be on the net :) Please let me know if you find something that I rather have removed.
The names of the files should be self explanatory what I tried to capture. If you believe a screensharing session would be beneficial, let me know :)

I have reduced the quality to the lowest values via the reolink app for fluent. Still same errors
![image](https://github.com/QuantumEntangledAndy/neolink/assets/7044850/8479fc15-4d45-4fbd-adf7-4e7bdc24dbf6)


[PFSense_neolink_captures.zip](https://github.com/QuantumEntangledAndy/neolink/files/12253397/PFSense_neolink_captures.zip)

**Comment by QuantumEntangledAndy** [comment]
There are very few packets in these dumps. Not enough to piece together what is happening. Please can you send the full process. From initial connection attempt up to the login that fails. Your log suggests you have a success discovery then fail at login, but all of that is missing from these dumps

**Comment by tr1plus** [comment]
Well. I quit neolink again before stopping the packet capture, so there are no more packets. Not sure why they aren't showing up. I believe that is exactly the issue occuring. Discovery works, but after that nothing happens and it tries to re-discover.

I'll try to install neolink on my old windows machine later today to rule out linux things I guess (since both docker & my linux VM run linux).

Kind regards

**Comment by QuantumEntangledAndy** [comment]
Thats the thing there are not enough packets in there even for discovery. There are only about 10 packets each they are basic udp related communications without details.

Take `PFSense_neolinkLinux_discoveryLocal_cameraCapture` as an example. There are 4 packets:

- Packet1:
```xml
<P2P>
<D2R_HB>
<uid>******</uid>
<dev>
<ip>192.168.1.79</ip>
<port>54185</port>
</dev>
<needrsp>0</needrsp>
<token>1688520982</token>
</D2R_HB>
</P2P>
```

- Packet 2:
```xml
<P2P>
<D2R_HB>
<uid>******</uid>
<token>1688520982</token>
<needrsp>1</needrsp>
</D2R_HB>
</P2P>
```

- Packet 3
```xml
<P2P>
<R2D_HB_R>
<rsp>0</rsp>
<time_t>1691082843</time_t>
<timer>
<hb>20000</hb>
</timer>
<retry/>
<debug>83886080</debug>
<up>1689780067</up>
</R2D_HB_R>
</P2P>
```
- Packet 4:
```xml
<P2P>
<D2R_HB>
<uid>****</uid>
<dev>
<ip>192.168.1.79</ip>
<port>54185</port>
</dev>
<needrsp>0</needrsp>
<token>1688520982</token>
</D2R_HB>
</P2P>
```

All of them are either D2R or R2D (reolink to device) or (device to reolink). Nothing involving the client. Please check your filter or the way you captured them

**Comment by tr1plus** [comment]
This is the capture, which I believe is as broad as I can go for anything going to/coming from 192.168.1.79.

<img width="586" alt="image" src="https://github.com/QuantumEntangledAndy/neolink/assets/7044850/769eddfb-f0c9-4a4c-a186-ea7536df5070">

I am willing to share with you a full packet capture without a "host address" filter, but I rather share it privately (discord/email/...) in that case since it will contain data across my whole subnet.
Unless you have an idea about what other filters I could set-up/remove.

Thanks!

**Comment by QuantumEntangledAndy** [comment]
If you want to filter it just to baichuan packets (the name of the camera protocol) you could use our dissector in https://github.com/QuantumEntangledAndy/neolink/blob/master/dissector/baichuan.lua to filter it in wireshark first. That will also let you see what is in there. Then you can reexport it from wireshark with the just the selected packets


<img width="1265" alt="Screenshot 2023-08-04 at 13 57 54" src="https://github.com/QuantumEntangledAndy/neolink/assets/13386481/46ba2ef2-8057-4aad-ba88-cd2e392141c2">

**Comment by QuantumEntangledAndy** [comment]
The whole subnet should not be necessary but broadcasts on `*.255` might be needed if you rely on local only discovery

**Comment by QuantumEntangledAndy** [comment]
Ok so

- discovery happens
- We send login request
- camera acknoledges our request
- we wait for camera to reply with next step of login
- we wait some more
- more waiting
- Camera drops us for inactivity

We are at the stage where we are waiting for data from the camera before we can do anything, but its not comming.

Can you send the dump from an offical client now? I will compare the two and see what is different

**Comment by tr1plus** [comment]
Hi

See attached.
Had to use a different computer since the reolink client doesn't want to start anymore on my old windows machine. I probably removed too many drives/.. already :)

192.168.1.50 is the "client" I'm using. I don't think I can select local discovery with the official client, so I assume it goes via reolink servers.

[official_client.zip](https://github.com/QuantumEntangledAndy/neolink/files/12259247/official_client.zip)

**Comment by QuantumEntangledAndy** [comment]
Can you test this [build](https://github.com/QuantumEntangledAndy/neolink/actions/runs/5760710609). Will fix the simulatneous subscription thing at least

**Comment by tr1plus** [comment]
still same error. I did another capture
[new_version.zip](https://github.com/QuantumEntangledAndy/neolink/files/12260194/new_version.zip)

**Comment by QuantumEntangledAndy** [comment]
Ok so:

Offical client:

f0debc0a01000000000000000000000012dc1465

Neolink:

f0debc0a010000002c0700000000000002dc1465323132333246323937413537413541373433383934413045344138303146430045313041444333393439424135394142424535364530353746323046383833000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000


All that extra junk is what we call the legacy login. It's used to login to the old protocol, its actually a huge secutrity risk since its MD5 hash of the usename and password.

It seems that newer clients skip it and go straight for what we call the modern login. 

I'll see if I can work something up

**Comment by tr1plus** [comment]
Okay, thanks a lot already for the efforts!

P.S. would you mind if I remove the zips with the captures again? Since they contain the UID & potentially some other info I rather not keep them on here :)
If needed, I can always re-share or re-create certain captures.

**Comment by tr1plus** [comment]
done. Still get the same error in console.
Attached new capture

[new_version2.zip](https://github.com/QuantumEntangledAndy/neolink/files/12261148/new_version2.zip)

**Comment by QuantumEntangledAndy** [comment]
Thanks lets see if its to do with the -1 ack packets. Please test this build

https://github.com/QuantumEntangledAndy/neolink/actions/runs/5763169512

**Comment by tr1plus** [comment]
still same error

New Capture: 
[new_version3.zip](https://github.com/QuantumEntangledAndy/neolink/files/12261741/new_version3.zip)

**Comment by QuantumEntangledAndy** [comment]
Alright [next one](https://github.com/QuantumEntangledAndy/neolink/actions/runs/5763527712), I'm gonna get some sleep here and try again in the morning. Maybe see if I can replicate by updating my firmware too

**Comment by QuantumEntangledAndy** [comment]
Hopefully its just not being recognised and the current AES will work with coercion anyways [next build](https://github.com/QuantumEntangledAndy/neolink/actions/runs/5763919389)

**Comment by tr1plus** [comment]
Hi

Thanks!
Same error in console.

New capture: 
[new_version5.zip](https://github.com/QuantumEntangledAndy/neolink/files/12263865/new_version5.zip)

P.S. I can maybe do some more testing tomorrow morning (Belgium timezone CET), but afternoon I'm not there. Sunday I will be there mostly full day too :)

**Comment by QuantumEntangledAndy** [comment]
next [build](https://github.com/QuantumEntangledAndy/neolink/actions/runs/5768479422)

**Comment by tr1plus** [comment]
Think we are able to login, but I get another error now:
![image](https://github.com/QuantumEntangledAndy/neolink/assets/7044850/cb30dfc1-f35e-4504-8fab-bfbd49e8b1fa)

Capture: 
[new_version6.zip](https://github.com/QuantumEntangledAndy/neolink/files/12266507/new_version6.zip)

Will also check in parallel if it's maybe something on my end, since it's a completely new error

**Comment by tr1plus** [comment]
![image](https://github.com/QuantumEntangledAndy/neolink/assets/7044850/2670e93a-f382-4686-9807-0246b4732e09)
This is what I get for both /tuin & /tuin/subStream

**Comment by tr1plus** [comment]
I saw another issue related to quality. As a reminder this are still my "fluent" quality settings. These are the lowest possible to select in the official client

![image](https://github.com/QuantumEntangledAndy/neolink/assets/7044850/8479fc15-4d45-4fbd-adf7-4e7bdc24dbf6)

**Comment by QuantumEntangledAndy** [comment]
The static is a dummy screen meaning that not enough frames have been sent yet. You need to wait until enough frames have arrived. Can you add the `stream = "subStream"` option to the config

**Comment by tr1plus** [comment]
Same result. when viewing /tuin/subStream I briefly got moving static, but it stopped after that again. /tuin just gave static.

**Comment by tr1plus** [comment]
Full documentation + trace of "subStream" option

```
bind = "0.0.0.0"

[[cameras]]
name = "tuin"
username = "admin"
password = "123456"
#address = "192.168.1.79"
uid = "xxx"
discover = "local"
stream = "subStream"
```

[new_version6_substream.zip](https://github.com/QuantumEntangledAndy/neolink/files/12266887/new_version6_substream.zip)

![image](https://github.com/QuantumEntangledAndy/neolink/assets/7044850/8e4449e5-aaf1-4705-aab3-113fb68d78aa)

**Comment by QuantumEntangledAndy** [comment]
That error means we received no video frame for 15s. If you leave neolink running do you ever get the buffers prepared message. Until you get the buffers prepared message you will always get static.

**Comment by tr1plus** [comment]
I will let it run for a while. I've had it run for ~5 min now and it keeps coming back with the same "warn" message as above.

Keep you posted :)

**Comment by tr1plus** [comment]
about 15minutes in. Keeps giving same message.
I did also reduce the quality of the "fluent" stream to lowest possible via the official app(so both quality options have the lowest possible quality drop down selections). But nothing changed in neolink.

**Comment by tr1plus** [comment]
30 min in - same story. I'll let it keep running till I have to leave in an hour.

Want me to test anything else or do another capture? I believe my last capture will be relevant but I'm happy to create a new one of needed :)

**Comment by QuantumEntangledAndy** [comment]
It's ok last capture should be fine. Just got to get the time to look at. Thanks for checking for so long.

**Comment by QuantumEntangledAndy** [comment]
So I looked at the dump and learned a few things. 1. There are some new fields in the meta data xml called 
```xml
<checkPos>0</checkPos>
<checkValue>177145857</checkValue>
```

Might be used for integrty checks


And! The actual video stream is encrypted! Never before have I seen reolink actually encrypt the video streams its always been in plain regradless of your password or where it was sent to... So it is a nice addition but also more work for me...

**Comment by tr1plus** [comment]
Glad to Read reolink is increasing their security 🤣.
Good luck on figuring this out. Let me know if/when I need to test anything.

And no worries. I should be thanking you for taking the time I figuring all this out!

**Comment by QuantumEntangledAndy** [comment]
Made some progress. I can get through login on the new method now and decode the first video frame, Then all I get are corrupt video frames. I am wondering if there is somekind of internal state being transmitted in the meta xml that hold the encryption intermediate variables. Will keep searching

**Comment by QuantumEntangledAndy** [comment]
Ok success, at least on my camera. Turns out they only encrypt SOME of the media frames.... Why? I don't know but there you go. Just have to read the extension xml first and see if `encryptLen` is set and decrypt jsut those first bytes.

**Comment by QuantumEntangledAndy** [comment]
Test build is [here](https://github.com/QuantumEntangledAndy/neolink/actions/runs/5776451800)

**Comment by tr1plus** [comment]
seems to be working via my windows machine now. I will try to set it up via docker too.

Quick question since I'm new to neolink. If I want to use mqtt I need to start with "mqtt" argument. How does that work via docker?
And does mqtt also allow rtsp?

**Comment by tr1plus** [comment]
seems to be working via docker too. Now time to test mqtt stuff. Let's hope they didn't change anything in the way you read out that data ;)

**Comment by tr1plus** [comment]
so it seems to be working in terms of RTSP. Thanks a lot already for that.

MQTT also seems to be working with the exception of the "motion" via PIR sensor.
![image](https://github.com/QuantumEntangledAndy/neolink/assets/7044850/84dee2eb-e4ff-43fb-81a6-5ca66e1d7f8c)

I do have the feeling that something is being triggered, as every time I move I see that a new MQTT message gets sent on the /status/motion, however the payload is "off".

Is there anything I can do do help on that one. Not sure if packet capture is useful in this case?

In the official client I have highest sensitivity
![image](https://github.com/QuantumEntangledAndy/neolink/assets/7044850/1e663187-ee9a-448e-aef6-d80b5effdeda)

**Comment by tr1plus** [comment]
P.S. - I also just made a donation trough ko-fi, but the message I wanted to share was too short :)

```
Thanks a lot for the work you are doing on neolink. I appreciate you taking the time in reverse-engineering reolink's firmware and make a feature available that should be there by default.
Thanks a lot for spending some time with me on github to deep-dive into this issue!

_I understand this small donation is in no-way representative of what you would have gotten as a normal pay-per-hour, I hope this allows you to keep maintaining neolink and add more amazing features!_
```

take care! hoping to have contributed my _very_ small part to this amazing project!

P.S. - let me know if you want me to open a new issue for the MQTT sensor thing.

P.S. 2 - could you share any insight if/when/how this will be merged in the main branch. I assume there might be a lot of regression issues because of the 2 different login & encryption mechanisms?

**Comment by QuantumEntangledAndy** [comment]
I'm planning to merge later today since. The regression should already mostly be handled since I tried to do that as I went. The only issue that I might have is with the old Swann cameras but I don't have any cameras to test with that anyways.

**Comment by QuantumEntangledAndy** [comment]
Also open a new issue with mqtt and check things out with mqtt explorer

**Comment by QuantumEntangledAndy** [comment]
If possible could you test run [this build](https://github.com/QuantumEntangledAndy/neolink/actions/runs/5780812549) it just some cleanup prior to a merge but I want to check that it doesn't break anything.

**Comment by tr1plus** [comment]
Seems something got broken

```
[2023-08-07T06:26:15Z INFO  neolink::mqtt::event_cam] tuin: Connecting to camera at Address: 192.168.1.79, UID: xxx
[2023-08-07T06:26:15Z INFO  neolink_core::bc_protocol] tuin: Trying TCP discovery
[2023-08-07T06:26:17Z INFO  neolink_core::bc_protocol] tuin: Trying local discovery
[2023-08-07T06:26:22Z WARN  neolink::mqtt::event_cam] Camera thread error: Failed to connect to camera tuin at Address: 192.168.1.79, UID: xxx on channel 0
    
    Caused by:
        Timed out while waiting for camera reply. Restarting...
```

Will start my windows PC in a bit and test there. with packet capture if needed.

**Comment by QuantumEntangledAndy** [comment]
Funnily enough I had the same issue this morning. Couldn't contact my battery cam and spend an ages going over the logs and wireshark trying to work it out only to realise the battery died last night... was one of those morning that starts "just right"

**Comment by tr1plus** [comment]
It's been mostly running fine I think.
MQTT did seem to have disconnected once, but that might have been caused because I was testing via another device in parallel. Not sure if/how it handels 2 devices connecting at the same time.
I have restarted MQTT docker 25 minutes ago and so far it's still running.

**Comment by QuantumEntangledAndy** [comment]
How well it handles multtple connections depends on the camera. Battery cameras seem to not work too well with muttiple connections. Probably a battery saving feature. Where as some the of the TCP ones can handles up to about 4 connections before they give up.

Gonna close this now as it should be now fixed in 0.5.17 feel free to reopen if not

**Comment by tr1plus** [comment]
FYI - just pulled the latest release via docker and so far it seems to be working. I'll let you know if I encounter any other issues.

Diving into some of the features neolink has to offer now :)

Thanks a lot!
